### ISSUE TITLE
Restrict FSN uniqueness to FSNs and preferred names for OpenMRS custom validation schema

### ISSUE THREAD

**Issue by bmamlin** [OpenConceptLab/ocl_issues#1921]
Currently, the OpenMRS custom validation schema on OCL is disallowing concepts with an FSN that matches any other non-retired name for any concept in the same locale. This uniqueness requirement should actually only apply across FSNs and _preferred_ names in the same locale (i.e., it is allowed for a concept to have a _non-preferred_ synonym that matches another concept's FSN in the same locale).

For CIEL v2024-09-04 (#1920) there are 41 concepts that failed import because they have an FSN that matches a _non-preferred_ name of another concept.

tl;dr – when applying name uniqueness constraint for the OpenMRS validation schema, limit uniqueness to non-retired FSNs and preferred synonyms within the same locale. If a synonym matches the FSN but the synonym is _not_ the preferred name, then the duplicate name should be allowed.

**Comment by snyaggarwal** [comment]
@bmamlin Excluding None and empty `name_type` along with SHORT and INDEX_TERM. 
This is deployed on QA and Staging.

**Comment by bmamlin** [comment]
> deployed on QA and Staging

Awesome! Hoping it can make it to production over the next week (before the next CIEL release, which I'm assuming will probably be in 1st or 2nd week of October). Thanks!

**Comment by snyaggarwal** [comment]
this was deployed everywhere, closing this.
